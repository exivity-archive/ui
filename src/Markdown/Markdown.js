import React from 'react'
import PropTypes from 'prop-types'
import ReactMarkdown from 'react-markdown'

import { List, Link, Paragraph, Heading } from 'reakit'

import Code from '../Code'
import CodeBlock from '../CodeBlock'
import Table from '../Table'

const defaultRenderers = {
  link: props => <Link {...props} />,
  paragraph: props => <Paragraph {...props} />,
  heading: props => <Heading {...props} as={`h${props.level}`} />,
  table: props => <Table {...props} />,
  inlineCode: props => <Code {...props} />,
  code: props => <CodeBlock {...props}>{props.value}</CodeBlock>,
  list: props => <List {...props} unordered={!props.ordered} />
}

const Markdown = ({ children, renderers }) => (
  <ReactMarkdown
    renderers={renderers}
    source={children} />
)

Markdown.propTypes = {
  children: PropTypes.string,
  renderers: PropTypes.object
}

Markdown.defaultProps = {
  renderers: defaultRenderers
}

export default Markdown
