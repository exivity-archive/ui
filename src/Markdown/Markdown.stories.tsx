import React from 'react'
import { storiesOf } from '@storybook/react'

import { ensureString } from '../utils'
// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
// @ts-ignore
import * as sample from '../utils/stories/samples/markdown.md'

import { Markdown } from '.'

storiesOf('atoms|Markdown', module)
  .add('default', () => <Markdown>{ensureString(sample)}</Markdown>)
