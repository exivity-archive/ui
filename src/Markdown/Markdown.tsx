import React, { ReactElement } from 'react'
import ReactMarkdown from 'react-markdown'

import { Code } from '../Code'
import { Heading } from '../Heading'
import { HeadingType } from '../Heading/Heading'
import { Link } from '../Link'
import { List } from '../List'
import { Paragraph } from '../Paragraph'
import { Table } from '../Table'
import { ObjectOf } from '../utils'

type Renderer = (props: any) => ReactElement

const headingLevelToType: {
  [level: number]: HeadingType
} = { 1: 'header', 2: 'section', 3: 'sub', 4: 'sub', 5: 'sub', 6: 'sub' }

const defaultRenderers: ObjectOf<Renderer> = {
  code: props => <Code block {...props}>{props.value}</Code>,
  heading: ({ level, ...rest }) => <Heading type={headingLevelToType[level]} {...rest} />,
  inlineCode: props => <Code {...props} />,
  link: props => <Link {...props} />,
  list: props => <List {...props} unordered={!props.ordered} />,
  paragraph: props => <Paragraph {...props} />,
  table: props => <Table {...props} />
}

interface MarkdownProps {
  renderers?: ObjectOf<Renderer>
  children?: string
}

export const Markdown = ({ children, renderers = {} }: MarkdownProps) => (
  <ReactMarkdown
    renderers={{
      ...defaultRenderers,
      ...renderers
    }}
    source={children} />
)
