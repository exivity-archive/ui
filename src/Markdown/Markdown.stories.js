import React from 'react'
import { storiesOf } from '@storybook/react'

import example from './stories/example.md'

import Markdown from './Markdown'

storiesOf('atoms|Markdown', module)
  .add('default', () => <Markdown>{example}</Markdown>)
