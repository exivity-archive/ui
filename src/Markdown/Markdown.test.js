/* eslint-disable no-undef */
import React from 'react'
import renderer from 'react-test-renderer'

import Markdown from './Markdown'

test('renders default Markdown', () => {
  const component = renderer.create(<Markdown />)
  expect(component.toJSON()).toMatchSnapshot()
})
