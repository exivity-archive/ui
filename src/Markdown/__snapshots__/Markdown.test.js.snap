// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default Markdown 1`] = `null`;
